### PR TITLE
Issue 1473/sort tag value

### LINKED ISSUE #1473: Sorting a view by the value of a value tag does not work
**Please be as thorough as possible, but if you can't answer everything, please just submit what you have! Thank you!**

## Description
It does not work to sort a view based on the values in a value tag column.

## Steps to reproduce
- Put a value tag on a person and give it a value
- Create a view based on a smart search that looks for this tag (all included people have the tag)
- put in a column that displays the tag
- highlight the column to sort by tag.

### Expected Behaviour
The view is sorted by what I write on the tag: numbers first, then alphabetically, then tags with no extra description. 

### Actual Behaviour
View remains unsorted.

### PR DESCRIPTION
**Pull request by kaulfield23**
## Description
This PR fixes a malfunctioning sort of person tags in view. When value is null, it will be sorted at the very top of the rows in ascending order.

## Screenshots

https://github.com/zetkin/app.zetkin.org/assets/77925373/73a8e5f5-3c19-4817-94aa-ac9d504bebbd

## Changes

* Adds `sortComparator` in `PersonTagColumnType`

## Related issues
Resolves #1473

### CODE REVIEW

File: src/features/views/components/ViewDataTable/columnTypes/PersonTagColumnType.tsx [hunk 1]
```diff
@@ -41,6 +41,39 @@ export default class PersonTagColumnType implements IColumnType {
...
+          if (!isNaN(Number(v1.value)) && !isNaN(Number(v2.value))) {
+            return parseInt(v1.value) - parseInt(v2.value);
+          }
+          if (isNaN(Number(v1.value))) {
+            return 1;
+          }
+          if (isNaN(Number(v2.value))) {
+            return -1;
+          } else {
+            v1.value.localeCompare(v2.value);
```
Comment (ziggabyte) [inline src/features/views/components/ViewDataTable/columnTypes/PersonTagColumnType.tsx:68]: Is it intentional to not return anything here? I would have expected this line to return the value of the `localeCompare`?

**Review by ziggabyte — COMMENTED** [review]
Having tested it, I think my main feedback is that it feels a bit strange that the first time I click the "sort" button, i only see the empty cells of the people that don't have the tag (or, for now, have the tag but don't have a value). I think a nicer user experience would be to have a null or empty value be treated as the "lowest" rather than the "highest", so that the first click on the sort button will return the rows with values at the top. 
Was this intentional or a consequence of some other desicions?

**Comment by kaulfield23** [comment]
> Yes! @ziggabyte  I also had a same thought, and checked another column to see how it behaves. `Notes` column for example, its order is "null - text" in ascending order and "text - null" in descending order. So I followed the same rule to here. Besides, it is not possible to check if `v1` and `v2`  are in asc, or desc order. So I created `v3` in parameter to see how it looks like. It can check the order from property `api` but it has this warning so it didn't feel so right. Therefore I decided to follow other column's pattern
image

File: src/features/views/components/ViewDataTable/columnTypes/PersonTagColumnType.tsx (hunk 1 above)
Comment (kaulfield23) [reply src/features/views/components/ViewDataTable/columnTypes/PersonTagColumnType.tsx:68]: Oops it is my mistake..!

**Comment by richardolsson** [comment]
> Yes! @ziggabyte I also had a same thought, and checked another column to see how it behaves. `Notes` column for example, its order is "null - text" in ascending order and "text - null" in descending order. So I followed the same rule to here.
> 
> Besides, it is not possible to check if `v1` and `v2` are in asc, or desc order. (Maybe if I add more properties in `ZetkinTag` then it is possible but I'm unsure of idea adding something to `ZetkinTag`) So I created `v3` in parameter to see how it looks like. It can check the order from property `api` but it has this warning so it didn't feel so right. Therefore I decided to follow other column's pattern
> 
> image

You were right to not make this change IMO. But I agree with @ziggabyte and would recommend we create another issue and tag it with design-needed, in order to think it through and make it consistent everywhere.

**Review by ziggabyte — APPROVED** [review]
Great job! :D

**Pull request merged**